overlay_test.go

```diff
@@ -181,3 +185,431 @@ series: bionic
 	sort.Strings(expErrors)
 	c.Assert(errStrings, jc.DeepEquals, expErrors)
 }
+
+func (*bundleDataOverlaySuite) TestReadAndMergeBundleData(c *gc.C) {
+	data := `
```

Comment:
Would this be easier to understand if we broke it into several tests? ReadAndMergeBundleDataChangeCharm
ReadAndMergeBundleDataChangeSeries, etc.?
it takes a lot for me to read through this test and really map all the things that the comments say into things that it is actually doing, and track that it isn't doing anything that *isn't* commented.

Thoughts?

Reply:
Yeah... I admit this one is a horribly verbose test but I wanted to have a test that checked all possible the mutations in a single block as it could help catch subtle bugs that don't manifest when checking each mutation in isolation.

Having said that, I don't mind to make the changes you propose especially if it makes reading the tests easier.